### PR TITLE
[gatsby-remark-images] allow user-specified max-width to overwrite presentationWidth

### LINKED ISSUE #15578: [gatsby-remark-images] Can't specify max-width
## Description

When using `gatsby-remark-images`, it is impossible to specify a custom `max-width` using the `wrapperStyle` option. The specified width will always be overwritten by the `presentationWidth` returned by `gatsby-plugin-sharp`.

This is problematic in cases where you want your `max-width` specified in non-pixel units. One such case is when you want your images to be fluid relative to the viewport height (used in the below examples).

A good starting point for debugging is [where the style is specified](https://github.com/gatsbyjs/gatsby/blob/0cefcb2f5a3ccf53cf15ea6b4fe2712809d134ca/packages/gatsby-remark-images/src/index.js#L274-L276)

### Steps to reproduce

Add `gatsby-remark-images` to your site, using a config specifying both `maxWidth` (required) and a `wrapperStyle` containing `max-width`. An example is:

```js
{
    resolve: `gatsby-remark-images`,
    options: {
        maxWidth: 1920,
        // intended to make image have max-height of 100vh - cannot use max-height as the placeholder depends on width
        wrapperStyle: fluid => `max-width:${(fluid.aspectRatio * 100).toFixed(2)}vh !important;`,
    }
}
```

### Expected result

The specified `max-width` is used (up for debate whether it should require `!important`)

### Actual result

`max-width` is always set to the `presentationWidth` returned by `gatsby-plugin-sharp` (in this case `1920px`).

### Environment

<details><summary>Result of <code>gatsby info</code></summary><pre><code>  System:
    OS: Linux 5.1 Arch Linux undefined
    CPU: (4) x64 Intel(R) Core(TM) i5-4200U CPU @ 1.60GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 11.15.0 - /usr/bin/node
    Yarn: 1.16.0 - /usr/bin/yarn
    npm: 6.9.2 - /usr/bin/npm
  Languages:
    Python: 3.7.3 - /usr/bin/python
  Browsers:
    Firefox: 67.0.4
  npmPackages:
    gatsby: ^2.3.25 => 2.13.4 
    gatsby-image: ^2.0.37 => 2.2.4 
    gatsby-plugin-glslify: ^1.0.0 => 1.0.0 
    gatsby-plugin-manifest: ^2.0.28 => 2.2.1 
    gatsby-plugin-offline: ^2.0.25 => 2.2.1 
    gatsby-plugin-prefetch-google-fonts: ^1.4.2 => 1.4.2 
    gatsby-plugin-react-helmet: ^3.0.12 => 3.1.0 
    gatsby-plugin-react-svg: ^2.1.1 => 2.1.1 
    gatsby-plugin-sass: ^2.0.11 => 2.1.0 
    gatsby-plugin-sharp: ^2.0.32 => 2.2.2 
    gatsby-plugin-styled-components: ^3.0.7 => 3.1.0 
    gatsby-remark-copy-linked-files: ^2.1.0 => 2.1.0 
    gatsby-remark-external-links: 0.0.4 => 0.0.4 
    gatsby-remark-images: ^3.1.3 => 3.1.3 
    gatsby-source-filesystem: ^2.0.29 => 2.1.2 
    gatsby-transformer-remark: ^2.3.8 => 2.6.0 
    gatsby-transformer-sharp: ^2.1.17 => 2.2.1 
</code></pre></details>

### PR DESCRIPTION
**Pull request by birjolaxew**
## Description

Currently gatsby-remark-images enforces a `max-width` on each image equal to the `presentationWidth` returned by gatsby-plugin-sharp. Any user-specified `max-width` in the `wrapperStyle` is simply ignored.  
Use case for custom `max-width` is described in #15578 

This PR adds support for specifying a `max-width` in the `wrapperStyle` which will be used instead of the `presentationWidth`.

## Related Issues

Fixes #15578

### CODE REVIEW

File: packages/gatsby-remark-images/src/index.js [hunk 1]
```diff
@@ -268,12 +268,15 @@ module.exports = (
...
+    const shouldApplyMaxWidth =
+      !imageCaption && wrapperStyle.indexOf(`max-width:`) !== -1
+
     rawHTML = `
     <span
       class="${imageWrapperClass}"
       style="position: relative; display: block; margin-left: auto; margin-right: auto; ${
         imageCaption ? `` : wrapperStyle
-      } max-width: ${presentationWidth}px;"
+      } ${shouldApplyMaxWidth ? `max-width: ${presentationWidth}px;` : ``}"
```
Comment (pieh) [inline packages/gatsby-remark-images/src/index.js]: What about moving `max-width` before ``${imageCaption ? `` : wrapperStyle}``? So `maxWidth` from user supplied `wrapperStyle` would overwrite default one.

This seems fragile - we have 2 different conditions relying on `imageCaption` and if we ever change this condition in future, it's very easy to miss it in one of the places

**Comment by wardpeet** [comment]
ping @birjolaxew any chance you can resolve this comment?

**Comment by birjolaxew** [comment]
@wardpeet @pieh 

Due to some changes in my personal life I unfortunately won't be updating this PR (although I agree with your suggestion, @pieh). If anyone feels like opening a new one for this issue, feel free.

**Comment by pieh** [comment]
@birjolaxew I hope this means change of focus and not any problems happening in your personal life! Even tho we didn't get to finish this changeset - I'd like to thank you for opening this PR ❤️

**Pull request closed without merging**